services/gitdiff/gitdiff.go

```diff
@@ -1497,6 +1514,68 @@ func GetDiff(gitRepo *git.Repository, opts *DiffOptions, files ...string) (*Diff
 	return diff, nil
 }
 
+// SyncAndGetUserSpecificDiff is like GetDiff, except that user specific data such as which files the given user has already viewed on the given PR will also be set
+// Additionally, the database asynchronously is updated if files have changed since the last review
+func SyncAndGetUserSpecificDiff(ctx context.Context, userID int64, pull *models.PullRequest, gitRepo *git.Repository, opts *DiffOptions, files ...string) (*Diff, error) {
+	diff, err := GetDiff(gitRepo, opts, files...)
+	if err != nil {
+		return nil, err
+	}
+	review, err := pulls.GetNewestReviewState(ctx, userID, pull.ID)
+	if err != nil || review == nil || review.UpdatedFiles == nil {
+		return diff, err
+	}
+
+	// Prepation to check for each file whether it was changed since the last review
+	changedFiles, err := gitRepo.GetFilesChangedBetween(review.CommitSHA, pull.HeadBranch)
```

Comment:
As noted on Discord, we cannot pass `pull.HeadBranch`, we need to pass something like `opts.AfterCommitID` into this. Currently PR's that have the head branch at another repo will error.